test/spec/modules/33acrossBidAdapter_spec.js

```diff
@@ -977,6 +992,40 @@ describe('33acrossBidAdapter:', function () {
       });
     });
 
+    context('when Global Placement ID (gpid) is defined', function() {
+      let bidderRequest;
+
+      beforeEach(function() {
+        bidderRequest = {};
+      });
+
+      it('passes the Global Placement ID (gpid) in the request', function() {
+        const ttxRequest = new TtxRequestBuilder()
+          .withBanner()
+          .withProductAndGPID('siab', 'fakeGPID0')
+          .build();
+        const serverRequest = new ServerRequestBuilder()
+          .withData(ttxRequest)
+          .build();
+
+        let copyBidRequest = utils.deepClone(bidRequests);
+        const addGPIDBidRequests = copyBidRequest.map(function(bidRequest, index) {
+          return {
+            ...bidRequest,
+            ortb2Imp: {
+              ext: {
+                gpid: 'fakeGPID' + index
+              }
+            }
+          };
+        });
+
+        const [ builtServerRequest ] = spec.buildRequests(addGPIDBidRequests, bidderRequest);
+
+        validateBuiltServerRequest(builtServerRequest, serverRequest);
```

Comment:
I am not seeing the expected field `fakeGPID<number` being added to the expected value `serverRequest`, this unit test is probably NOT failing because `validateBuiltServerRequest` doesn't run assertions on the `ext` field. Please change it accordingly .

Reply:
In the validateBuiltServerRequest function, I see this line
expect(JSON.parse(builtReq.data)).to.deep.equal(JSON.parse(expectedReq.data) -
For serverRequest and builtServerRequest, I see the gpid field with value fakeGPID0 passed in and is stored under the data field
```
{
"data": "{"imp":[{"id":"b1","banner":{"format":[{"w":300,"h":250},{"w":728,"h":90}],"ext":{"ttx":{"viewability":{"amount":100}}}},"ext":{"ttx":{"prod":"siab"},"gpid":"fakeGPID0"}}],"site":{"id":"sample33xGUID123456789"},"id":"r1","regs":{"ext":{"gdpr":0}},"ext":{"ttx":{"prebidStartedAt":1,"caller":[{"name":"prebidjs","version":"7.6.0pre"}]}}}"
"method": "POST"
"options": {
"contentType": "text/plain"
"withCredentials": true
}
"url": "https://ssc.33across.com/api/v1/hb?guid=sample33xGUID123456789"
}
```
Would validateBuiltServerRequest be ok to use in this case? Let me know if you see something different

Reply:
my bad, you are right, that assertion does the validation.